packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel.TestProjects/Unbranded-TypeSpec/src/Generated/Models/Thing.cs

```diff
@@ -27,7 +27,7 @@ public partial class Thing
         public string RequiredLiteralFloat { get; set; } = "1.23";
```

Comment:
The IsLiteral property of CSharpType is incorrectly returning false for int and float. We have a issue to track this - https://github.com/Azure/autorest.csharp/issues/4630.

Reply:
I am trying to fix it in this PR: https://github.com/microsoft/typespec/pull/3224 (despite its name does not mention this, it is still WIP)